indico/modules/announcement/templates/display.html

```diff
@@ -1,23 +1,80 @@
+            {% if not persistent %}
+                <span class="icon-close close-button"></span>
+            {% endif %}
+        </div>
+    {% endif %}
+{% endmacro %}
 
-        var hash = {{ message | crc32 | string | tojson }};
+{% macro render_announcement_new(type, message, persistent=false) %}
+    {% if type == 'html' %}
+        {{ message | safe }}
+    {% elif type == 'logged-in-as' %}
+    {% else %}
+        <div class="announcement-bar weak-hidden {{type}}" data-hash={{message | crc32 | string | tojson}}>
+            <div class="content">
+                <i class="warning sign large icon"></i>
+                <div class="message">
+                    {{ message }}
+                </div>
+                <i class="close icon close-button"></i>
+            </div>
+        </div>
+    {% endif %}
+{% endmacro %}
 
-        $('#announcement-bar .close-button').on('click', function() {
-            $.jStorage.set('hideAnnouncement', hash);
-            $('#announcement-bar').slideUp('fast');
-        });
+{% macro render_announcements(legacy=false) %}
+    {% set announcements = template_hook('global-announcement', as_list=true) %}
+    {% if announcements %}
+        {% for announcement in announcements %}
+            {% if legacy %}
+                {{ render_announcement_legacy(*announcement) }}
+            {% else %}
+                {{ render_announcement_new(*announcement) }}
+            {% endif %}
+        {% endfor %}
+        <script>
+            function refreshPageHeight() {
+                let numBars = Array.from(document.querySelectorAll('.announcement-bar'))
+                    .map(function(e) { return e.style.display === 'flex' })
+                    .filter(function(e) { return !!e; })
+                    .length;
 
-        if ($.jStorage.get('hideAnnouncement') !== hash) {
-            $("#announcement-bar").show();
-        }
-    })();
-</script>
+                if (document.querySelector('.logged-in-as-bar')) {
+                    numBars++;
+                }
+
+                document.querySelector('body').style.setProperty('--extra-bars', numBars);
+            }
+
+            (() => {
+                document.querySelectorAll('.announcement-bar .close-button').forEach(function(elem) {
+                    elem.addEventListener('click', function() {
+                        const bar = elem.closest('.announcement-bar');
+                        window.localStorage.setItem('hideAnnouncement', bar.dataset.hash);
+                        bar.style.display = 'none';
+                        refreshPageHeight();
+                    });
+                });
+
+                document.querySelectorAll('.announcement-bar').forEach(function(elem) {
+                    if (window.localStorage.getItem('hideAnnouncement') !== elem.dataset.hash) {
+                        elem.style.display = 'flex';
+                    }
+                });
+                refreshPageHeight();
+
```

Comment:
:white_large_square:

Reply:
![image](https://user-images.githubusercontent.com/2699/52581171-58831300-2e2a-11e9-827a-44423b3c5d1a.png)

Is this a smiley or something?

Reply:
Extra white line I guess he means

Reply:
Oh, I thought it was a smiley that wasn't rendering in here 😅
Thanks @mvidalgarcia!